druid/widget/navigation_handler/navigation_handler.lua

```diff
@@ -0,0 +1,442 @@
+-- Title: Navigation Handler
+-- Description: Adds support for navigating with keyboard and gamepad.
+-- Author: NaakkaDev <https://github.com/NaakkaDev>
+-- Widget: navigation_handler
+-- Tags: input
+-- Version: 1
+
+
+local event = require("event.event")
+local const = require("druid.const")
+
+
+---Widget force handling GUI navigation via keyboard/gamepad.
+---
+---### Setup
+---Loads the widget module:
+---`local navigation_handler = require("druid.widgets.navigation_handler.navigation_handler")`
+---
+---Create the new widget instance:
+---`self.nav = self.druid:new_widget(navigation_handler, self.my_button)`
+---
+---
+---### Example using the `on_select` event
+---```
+---local function on_select_btn(self, new, current)
+---    gui.play_flipbook(new.node, "button_selected")
+---    gui.play_flipbook(current.node, "button")
+---end
+---```
+---With `self.nav.on_select:subscribe(on_select_btn)`
+---
+---
+---### Notes
+---- `on_select` event callback params: (self, component_instance, component_instance).
+----   - **self** - Druid self context.
+----   - **new** - The component that will be selected next.
+----   - **current** - The component that is about to be de-selected.
+---- Key triggers in `input.binding` should match your setup.
+---- Used `action_id`'s are:' `key_up`, `key_down`, `key_left` and `key_right`.
+---@class druid.widget.navigation_handler: druid.widget
+---@field on_select event fun(self, component_instance, component_instance) Triggers when a new component is selected. The first component is for the newly selected and the second is for the previous component.
+---@field private _weight number The value used to control of the next button diagonal finding logic strictness.
+---@field private _tolerance number Determines how lenient the next button finding logic is. Set larger value for further diagonal navigation.
+---@field private _select_trigger hash Select trigger for the current component. Defaults to `druid.const.ACTION_SPACE`.
+---@field private _selected_triggers table Table of action_ids that can trigger the selected component. Valid only for the current button when set.
+---@field private _selected_component druid.component|druid.button|druid.slider Currently selected button instance.
+---@field private _deselect_directions table<string> The valid "escape" direction of the current selection.
+local M = {}
+
+-- Components that support navigation.
+local COMPONENTS = { "button", "slider" }
+
+
+---The constructor for the navigation_handler component.
+---@param component druid.component Current druid component that starts as selected.
+---@param tolerance number|nil How far to allow misalignment on the perpendicular axis when finding the next component.
+function M:init(component, tolerance)
```

Comment:
Let's get rig from args from constructor, to always have simple code init for any widget
`new_widget(navigation_handler)`

The component can be set right after with kind of `select_component` and tolerance already has a default value and can be changed right after if required

Reply:
Makes sense! Added a nil check for the `_selected_component` at the start of the on_input() also.

Btw, which one of use is supposed to click that "Resolve conversation" button? I have never used this github feature before.